yt/yt/server/scheduler/scheduler.cpp

```diff
@@ -2659,6 +2660,31 @@ class TScheduler::TImpl
             return;
         }
 
+        if (operation->Spec()->IssueTemporaryToken) {
+            try {
+                // We check this first, so that we don't create a token in vain.
+                auto secureVault = operation->GetSecureVault();
+                if (secureVault && secureVault->FindChild(operation->Spec()->TemporaryTokenEnvironmentVariableName)) {
+                    THROW_ERROR_EXCEPTION(
+                        "Temporary token environment variable %Qv already exists in secure vault",
+                        operation->Spec()->TemporaryTokenEnvironmentVariableName);
+                }
+
+                auto temporaryToken = WaitFor(MasterConnector_->IssueTemporaryOperationToken(operation))
+                    .ValueOrThrow();
+                // NB: This modifies secure vault contents.
+                operation->SetTemporaryToken(temporaryToken.Token);
+                operation->SetTemporaryTokenNodeId(temporaryToken.NodeId);
```

Comment:
Maybe join this methods into one?

Reply:
Ok, moved the set into the previous method with an extra parameter. Do not want to introduce dependencies on `TIssueTokenResult` into `operation.h` though, it would be weird.